Source/Core/Core/PowerPC/Interpreter/Interpreter_LoadStorePaired.cpp

```diff
@@ -296,12 +298,12 @@ void Interpreter::Helper_Dequantize(u32 addr, u32 instI, u32 instRD, u32 instW)
     break;
   }
 
-  if (PowerPC::ppcState.Exceptions & EXCEPTION_DSI)
+  if (ppcs->Exceptions & EXCEPTION_DSI)
   {
     return;
   }
 
-  rPS(instRD).SetBoth(ps0, ps1);
+  ppcs->ps[instRD].SetBoth(double(ps0), double(ps1));
```

Comment:
I'm not sure if this cast is correct. Both ps0 and ps1 may contain any arbitory binary data for QUANTIZE_FLOAT, and if fact, some games use this for memcpy...

Reply:
I'm pretty sure the old code did the cast too, just implicitly (though I see your point, especially for NaNs). The thing is, we need to convert the values to double representation since that's what's required for vectorization apparently. Any ideas how to do it better?

Reply:
Do you mean the compiler doesn't autovectorize if the code implicitly converts to doubles?

Reply:
https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/Core/PowerPC/Interpreter/Interpreter_FPUtils.h#L392

Reply:
But such a change should not be done within a cleanup PR. May you just drop the redundant cast here, and we'll fix this in the next PR?

Reply:
fwiw, this implicit conversion from float→double [predates](https://github.com/dolphin-emu/dolphin/blob/52cae18b016750b7e323fae6ff9b2bb3316cee1c/Source/Core/Core/PowerPC/Interpreter/Interpreter_LoadStorePaired.cpp#L304) the use of the paired single helper structure:

Reply:
I'm not sure the PPC float-double conversion algorithm (ConvertToDouble) even applies here, since PS0 and PS1 are only 32-bit in hardware (though PS0 can be used as input to double-precision operations apparently). Does the C++ standard have some round-tripping guarantees for float-double-float conversions? That would be all that has to be guaranteed for the memcpy case, right? Anyway, I agree that this PR is probably not the right place to change the conversion semantics. Is there any place to leave this as a TODO? (comment above the line? bug tracker?)

@shuffle2 I was referring to https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/Core/PowerPC/PowerPC.h#L141

Reply:
Well, I don't understand the addition of the explicit casts, it just makes the code more ugly.

Reply:
They have been removed from this PR, and it is merged now. The behavior change I've suggested is implemented in #7981.